### PR TITLE
add kache cli part 1(init cli)

### PR DESCRIPTION
**Pull request by Chyroc**
## PR

#58 

* [x] help message
* [ ] connect to server(tcp)
* [x] read input text
* [ ] parse input text to req command
* [ ] send input to kache server and read resp
* [ ] parse resp text to resp message
* [ ] render resp from server
  * [ ] get
  * [ ] set
  * [ ] exists
  * [ ] del
  * [ ] incr
  * [ ] decr
## Description

* add dep: github.com/c-bata/go-prompt
* init client

image

### CODE REVIEW

**Comment by kasvith** [comment]
We must use Cobra, check `kache`

**Comment by Chyroc** [comment]
need a interactive prompts, not just a command client, can Cobra do this?

**Comment by kasvith** [comment]
Cobra is need for take args and start the CLI, after that we can use normal code, but passing args should use Cobra like kache server

**Comment by kasvith** [comment]
add this commands to cobra-cmds/cli

**Comment by Chyroc** [comment]
done

File: cmd/kache-doc-gen/main.go [hunk 1]
```diff
@@ -29,12 +29,13 @@ import (
...
 	"github.com/spf13/cobra/doc"
 
+	"github.com/kasvith/kache/internal/cobra-cmds"
 	"github.com/kasvith/kache/internal/cobra-cmds/kache"
 )
 
 func main() {
 	// import all commands
-	kache.RootCmd.AddCommand(kache.VersionCmd)
+	kache.RootCmd.AddCommand(cobracmds.VersionCmd)
```
Comment (kasvith) [inline cmd/kache-doc-gen/main.go]: this is a utility so i hope to exclude this from build, please revert to original :)

File: internal/cobra-cmds/cli/cli.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	Use:   "kache-cli",
+	Short: "kache-cli is a client to access kache server",
+	Run:   runCli,
+}
+
+func init() {
+	RootCmd.Flags().StringP("host", "", "127.0.0.1", "host of kache server")
+	RootCmd.Flags().IntP("port", "p", 7088, "port of kache server")
+
+	viper.BindPFlag("port", RootCmd.Flags().Lookup("port"))
```
Comment (kasvith) [inline internal/cobra-cmds/cli/cli.go]: for CLI viper is not needed, it's used for parse TOML files with Cobra

File: internal/cobra-cmds/cli/cli.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	RootCmd.AddCommand(cobracmds.VersionCmd)
+
+	if err := RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func runCli(cmd *cobra.Command, args []string) {
+	host := viper.Get("host").(string)
```
Comment (kasvith) [inline internal/cobra-cmds/cli/cli.go]: we can use cobra directly IMO

File: internal/cobra-cmds/cli/cli.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	port := viper.Get("port").(int)
```
Comment (kasvith) [inline internal/cobra-cmds/cli/cli.go]: same in here

File: cmd/kache-doc-gen/main.go (hunk 1 above)
Comment (Chyroc) [reply cmd/kache-doc-gen/main.go]: remove this ? or move `kache.VersionCmd` to `cobracmds.VersionCmd`

or 

change `VersionCmd` to 
```go
var VersionCmd = &cobra.Command{
	Use:   "version",
	Short: "Display application version",
	Long:  `Display application version on the screen`,
	Run: func(cmd *cobra.Command, args []string) {
		if BuildTime == "" || GitHash == "" {
			fmt.Printf("Version: %s\n", APPVER)
			return
		}
		fmt.Printf("Version: %s, BuildTime: %s, GitHash: %s\n", APPVER, BuildTime, GitHash)
	},
}
```

Comment (kasvith) [reply cmd/kache-doc-gen/main.go]: just revert the file to original

Comment (Chyroc) [reply cmd/kache-doc-gen/main.go]: but i move `github.com/kasvith/kache/internal/cobra-cmds/kache. VersionCmd` to `github.com/kasvith/kache/internal/cobra-cmds. VersionCmd`

to let cli reuse it.

copy a `VersionCmd` from kache to cli package?

File: internal/cobra-cmds/version.go [hunk 5]
```diff
@@ -22,7 +22,7 @@
  * SOFTWARE.
  */
 
-package kache
+package cobracmds
```
Comment (kasvith) [inline internal/cobra-cmds/version.go]: this should be reverted, we have separate versions for server and cli. Also fix consecutive files.

File: cmd/kache-doc-gen/main.go (hunk 1 above)
Comment (kasvith) [reply cmd/kache-doc-gen/main.go]: sorry ignore this. build script will do versioning. if in future if we want to add separate info for either of binaries we can split them to two files

**Comment by Chyroc** [comment]
done

**Pull request merged**